lecturework.html

```diff
@@ -0,0 +1,119 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <title>Lesson 8 - Loops</title>
+  </head>
+
+  <body>
+    <div>
+      <h3>List of Products</h3>
+      <div>
+        <span>Iphone</span><br />
+        <span>Price: 900 €</span>
+        <input
+          type="button"
+          id="addPhoneToCart"
+          value="Add to shopping cart"
+        />
+      </div>
+      <br />
+      <div>
+        <span>Nintendo DS</span><br />
+        <span>Price: 1200 €</span>
+        <input
+          type="button"
+          id="addNintendoToCart"
+          value="Add to shopping car"
+        />
+      </div>
+      <br />
+      <div>
+        <span>MacBook Air</span><br />
+        <span>Price: 3200 €</span>
+        <input
+          type="button"
+          id="addMacbookToCart"
+          value="Add to shopping car"
+        />
+      </div>
+    </div>
+    <hr />
+    <div>
+      <h3>Shopping Cart</h3>
+      <input
+        type="button"
+        id="showShoppingCart"
+        value="Show Shopping Cart Products"
+      />
+      <div id="shoppingCart">
+        <!-- Display shopping cart products-->
+      </div>
+      <h3>
+        Total Price
+      </h3>
+
+      <span id="total">
+        <!-- show total price-->
+      </span>
+      <input type="button" id="totalPrice" value="Show total price" />
+    </div>
+
+    <script>
+
+      const phoneToCartEl = document.getElementById("addPhoneToCart");
+      const nintendoToCartEl = document.getElementById("addNintendoToCart");
+      const macbookToCartEl = document.getElementById("addMacbookToCart");
+      const showShoppingCartEl = document.getElementById("showShoppingCart");
+      const totalPriceEl = document.getElementById("total");
+
+      function addIphoneToShoppingCart() {
+        const product = {
+          name: "Iphone",
+          price: 900,
+      }
+      shoppingCart.push(product)
+      }
+
+      function addNintendoToShoppingCart() {
+        const product = {
+          name: "Nintendo DS", 
+          price: 1200,
+      }
+      shoppingCart.push(product)
+      }
+
+      function addMacbookToShoppingCart() {
+        const product = {
+          name: "Mac Book Air",
+          price: 3200,
+        }
+        shoppingCart.push(product)
+      };
+      function showShoppingCartProducts() {
+        const shoppingCartContainer = document.getElementById("shoppingCart");
+        for (let i=0; i < shoppingCart.length; i++) {
+          const productContainer = document.createElement("div");
+          productContainer.innerHTML = `
+          ${shoppingCart[i].name}<br>
+          ${shoppingCart[i].price}
+          `;
+          shoppingCartContainer.appendChild(productContainer)
+        }
+      };
+      function showTotalPrice() {
+        let total = 0;
+        for (let i = 0; i < total.length; i++){
```

Comment:
Here `total.length` is not correct, you should iterate over your shoppingCart array because in there you have you products and then access to the price of each product.

```
HINT: 
shoppingCart.length 
shoppingCart[0].price
```

Reply:
I changed it to this:
function showTotalPrice() {
        let total = 0;
        for (let i = 0; i < shoppingCart.length; i++){
          sum += shoppingCart[0].price;
        }
      };
but it's still not working. I've tried playing around but so far no luck.

Also, one more note. I didn't manage to make a note of how to clear the shopping cart. Would you be able to remind me please or do I find it in your ReDi repo? (will check out now). Thanks!

Reply:
`sum` variable does not exist, you should use total variable that you define before the for loop. in total you are saving the sum of prices.

Reply:
Thanks, resolved.